app/views/devise/sessions/_return_to_service_provider.html.slim

```diff
@@ -1,4 +1,5 @@
-.sm-col.py1.sm-p0
-  = link_to "‹ #{t('links.back_to_sp', sp: decorated_session.sp_name)}",
-    decorated_session.sp_return_url,
-    class: 'inline-block truncate align-bottom sm-maxw-190p'
+- if decorated_session.sp_return_url.present?
```

Comment:
I moved the conditional logic to the decorator.  The two partials are similar in that they both contain a link back to the SP but the formatting surrounding the links is slightly different so I opted to make two separate files.